app/src/main/java/com/yacgroup/yacguide/LaunchActivity.kt

```diff
@@ -59,8 +59,9 @@ class LaunchActivity : AppCompatActivity() {
     }
 
     private fun _switchToDatabase() {
-        val defaultRegionId = _customSettings.getInt(getString(R.string.default_region_key), R.integer.unknown_id)
```

Comment:
This was the bug: Accessing `R.integer.unknown_id` which does not represent the value itself but rather the id of the according xml resource.

Reply:
Why was this not a problem for higher AP versions?

Reply:
The problem is that accessing raw resource `R.integer.unknown_id` twice seems to give different results for lower API levels.

Reply:
I don't think it's API level related. I guess, I (or my emulator, respectively) had an older Yacguide version installed before, in which the raw resource id R.integer.unknown_id was different from a new app version. But since the old one was stored and never reset to -1, the if branch compared two different resource ids which led to the behavior that the app wants to load a region with an invalid id as start-up region.